File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.dart
class Solution {
  int findLHS(List<int> nums) {
    Map<int, int> map = {};
    int maxLen = 0;

    for (int i = 0; i < nums.length; ++i) {
      map[nums[i]] = (map[nums[i]] ?? 0) + 1;
    }
    for (int key in map.keys) {
      if (map.containsKey(key + 1)) {
        int currentLen = map[key]! + map[key + 1]!;
        maxLen = currentLen > maxLen ? currentLen : maxLen;
      }
    }
    return maxLen;
  }
}

File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.swift
// Print "Try programiz.pro" message

class Solution {
    func findLHS(_ nums: [Int]) -> Int {
      var maxLen = 0;
      var frequencyMap: [Int:Int] = [:]
      for number in nums{
          frequencyMap[number, default:0] += 1
      }
      for key in frequencyMap.keys{
          if let nextFreq = frequencyMap[key+1]{
              let currentLen = frequencyMap[key]! + nextFreq
              maxLen = max(maxLen, currentLen)
          }
      }
      print(frequencyMap)
    return maxLen
    }
}

// let here as final
// let solution =  Solution()
// print(solution.findLHS([1,3,2,2,5,2,3,7]))